pkg/controllers/deprovisioning/multimachineconsolidation.go

```diff
@@ -75,37 +75,37 @@ func (m *MultiMachineConsolidation) ComputeCommand(ctx context.Context, candidat
 func (m *MultiMachineConsolidation) firstNMachineConsolidationOption(ctx context.Context, candidates []*Candidate, max int) (Command, error) {
 	// we always operate on at least two machines at once, for single machines standard consolidation will find all solutions
 	if len(candidates) < 2 {
-		return Command{action: actionDoNothing}, nil
+		return Command{}, nil
 	}
 	min := 1
 	if len(candidates) <= max {
 		max = len(candidates) - 1
 	}
 
-	lastSavedCommand := Command{action: actionDoNothing}
+	lastSavedCommand := Command{}
 	// binary search to find the maximum number of machines we can terminate
 	for min <= max {
 		mid := (min + max) / 2
 
 		candidatesToConsolidate := candidates[0 : mid+1]
 
-		action, err := m.computeConsolidation(ctx, candidatesToConsolidate...)
+		cmd, err := m.computeConsolidation(ctx, candidatesToConsolidate...)
 		if err != nil {
 			return Command{}, err
 		}
 
 		// ensure that the action is sensical for replacements, see explanation on filterOutSameType for why this is
 		// required
-		if action.action == actionReplace {
-			action.replacements[0].InstanceTypeOptions = filterOutSameType(action.replacements[0], candidatesToConsolidate)
-			if len(action.replacements[0].InstanceTypeOptions) == 0 {
-				action.action = actionDoNothing
-			}
+		instanceTypeFiltered := false
```

Comment:
I'm somewhat curious why you want to make this change in the logic. I kinda think that the current way it's written makes a lot of sense

Reply:
We no longer have an action variable in the Command to artificially set it to DoNothing now, so this is the way to do it.

Reply:
Ah, got it. I guess this feels like indirect. What if we either set the max variable directly here _or_ created a `failedConsolidation` flag so it's more clear that the two states are:
1. Succeeded consolidation simulation (check upper half)
2. Failed consolidation simulation (check bottom half)